### PR TITLE
bump gotest.tools v3.0.1 for compatibility with Go 1.14

### PR DESCRIPTION
**Pull request by thaJeztah**
full diff: https://github.com/gotestyourself/gotest.tools/compare/v2.3.0...v3.0.1

relates to https://github.com/moby/moby/pull/40353

### CODE REVIEW

**Comment by thaJeztah** [comment]
Added "HOLD"; probably easier to merge https://github.com/moby/moby/pull/40473 first, as that one will conflict with this one

**Comment by thaJeztah** [comment]
Hm; looks like something went wrong re-vendoring;

```

[2020-02-07T14:11:37.806Z] The result of vndr differs
[2020-02-07T14:11:37.806Z] 
[2020-02-07T14:11:37.806Z]  D vendor/gotest.tools/LICENSE
[2020-02-07T14:11:37.806Z]  D vendor/gotest.tools/README.md
[2020-02-07T14:11:37.806Z]  D vendor/gotest.tools/go.mod
```

I'll remove them manually

**Comment by thaJeztah** [comment]
@cpuguy83 @tiborvass @AkihiroSuda this should be ready for review

**Review by AkihiroSuda — APPROVED** [review]

**Review by cpuguy83 — APPROVED** [review]
LGTM

File: pkg/authorization/api_test.go [hunk 1]
```diff
@@ -11,13 +11,13 @@ import (
...
-	"gotest.tools/assert"
-	is "gotest.tools/assert/cmp"
+	"gotest.tools/v3/assert"
+	is "gotest.tools/v3/assert/cmp"
 )
 
 func TestPeerCertificateMarshalJSON(t *testing.T) {
 	template := &x509.Certificate{
-		IsCA:                  true,
+		IsCA: true,
```
Comment (AkihiroSuda) [inline pkg/authorization/api_test.go]: Why? 
Seems causing goimports error: https://github.com/moby/moby/pull/40501

Comment (thaJeztah) [reply pkg/authorization/api_test.go]: Ugh, sorry about that; I suspect this has been my IDE formatting it (perhaps different go version?)

Also curious why CI didn't catch it

**Pull request merged**